daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/Ref.scala

```diff
@@ -47,6 +47,18 @@ object Ref {
   type ApplicationId = LedgerString
   val ApplicationId: LedgerString.type = LedgerString
 
+  /** Identifiers for participant node users, which act as clients to the Ledger API.
+   *
+   * The concept of participant node users has been introduced after the concept
+   * of client applications, which is why the code contains both of them.
+   * The [[UserId]] identifiers for participant node users are a strict subset of the
+   * [[ApplicationId]] identifiers for Ledger API client applications.
+   * The Ledger API and its backing code use [[UserId]] where possible
+   * and [[ApplicationId]] otherwise.
+   * */
+  type UserId = IdString.UserId
+  val UserId: IdString.UserId.type = IdString.UserId
```

Comment:
If the concept of user never makes it into the evaluation of commands, I don't think that `daml-lf` is a good place for this.

Reply:
Same thing applies to `ApplicationId` which has been in here for ages. I’m fine with keeping all of them in here.